diff --git c/configure.ac w/configure.ac
index 74284add87b5..1a257d64b173 100644
--- c/configure.ac
+++ w/configure.ac
@@ -77,6 +77,11 @@ AC_C_RESTRICT
 m4_include([config/m4/pkg.m4])
 PKG_PROG_PKG_CONFIG
 
+AC_ARG_ENABLE([embedded],
+             [AS_HELP_STRING([--enable-embedded],
+                             [Enable embedded mode @<:@default=no@:>@])
+             ])
+AM_CONDITIONAL([EMBEDDED], [test x"$enable_embedded" = x"yes"])
 
 #
 # Save config flags for version dump tool
diff --git c/src/ucm/Makefile.am w/src/ucm/Makefile.am
index 2215ddd21117..d5a5932674da 100644
--- c/src/ucm/Makefile.am
+++ w/src/ucm/Makefile.am
@@ -24,8 +24,10 @@ libucm_la_CFLAGS = \
 	$(BASE_CFLAGS) \
 	$(CFLAGS_NO_DEPRECATED)
 
+if !EMBEDDED
 nobase_dist_libucm_la_HEADERS = \
 	api/ucm.h
+endif
 
 noinst_HEADERS = \
 	event/event.h \
diff --git c/src/ucp/Makefile.am w/src/ucp/Makefile.am
index 5a2bb6a6f4e7..f96a3f488906 100644
--- c/src/ucp/Makefile.am
+++ w/src/ucp/Makefile.am
@@ -13,11 +13,13 @@ libucp_la_LDFLAGS  = -ldl -version-info $(SOVERSION)
 libucp_la_LIBADD   = ../ucs/libucs.la ../uct/libuct.la
 libucp_ladir       = $(includedir)/ucp
 
+if !EMBEDDED
 nobase_dist_libucp_la_HEADERS = \
 	api/ucp_compat.h \
 	api/ucp_def.h \
 	api/ucp_version.h \
 	api/ucp.h
+endif
 
 noinst_HEADERS = \
 	amo/amo.inl \
diff --git c/src/ucs/Makefile.am w/src/ucs/Makefile.am
index de930eeebefa..6b930964009f 100644
--- c/src/ucs/Makefile.am
+++ w/src/ucs/Makefile.am
@@ -17,6 +17,7 @@ libucs_la_LIBADD   = \
 	$(LIBM) \
 	$(top_builddir)/src/ucm/libucm.la
 
+if !EMBEDDED
 nobase_dist_libucs_la_HEADERS = \
 	algorithm/crc.h \
 	algorithm/qsort_r.h \
@@ -55,6 +56,7 @@ nobase_dist_libucs_la_HEADERS = \
 	type/status.h \
 	type/thread_mode.h \
 	type/cpu_set.h
+endif
 
 noinst_HEADERS = \
 	datastruct/arbiter.h \
diff --git c/src/uct/Makefile.am w/src/uct/Makefile.am
index 930f73b9b8ba..9a8f988ef495 100644
--- c/src/uct/Makefile.am
+++ w/src/uct/Makefile.am
@@ -15,11 +15,13 @@ libuct_la_LDFLAGS  = -ldl -version-info $(SOVERSION)
 libuct_la_LIBADD   = $(LIBM) ../ucs/libucs.la
 libuct_ladir       = $(includedir)/uct
 
+if !EMBEDDED
 nobase_dist_libuct_la_HEADERS = \
 	api/tl.h \
 	api/uct_def.h \
 	api/uct.h \
 	api/version.h
+endif
 
 noinst_HEADERS = \
 	base/addr.h \
