diff --git c/Makefile.am w/Makefile.am
index b92ff287de8c..21878c177ab9 100644
--- c/Makefile.am
+++ w/Makefile.am
@@ -21,13 +21,13 @@ DIST_SUBDIRS = $(SUBDIRS)
 
 # Only install the pkg file if we're building in standalone mode (and not on Windows)
 if HWLOC_BUILD_STANDALONE
-pkgconfigdir = $(libdir)/pkgconfig
-pkgconfig_DATA = hwloc.pc
+# pkgconfigdir = $(libdir)/pkgconfig
+# pkgconfig_DATA = hwloc.pc
 endif
 
 # Only install the valgrind suppressions file if we're building in standalone mode
 if HWLOC_BUILD_STANDALONE
-dist_pkgdata_DATA = contrib/hwloc-valgrind.supp
+# dist_pkgdata_DATA = contrib/hwloc-valgrind.supp
 endif
 
 #
