diff --git c/src/topology-xml-libxml.c w/src/topology-xml-libxml.c
index ac20d87f21ca..f35eab2795f1 100644
--- c/src/topology-xml-libxml.c
+++ w/src/topology-xml-libxml.c
@@ -134,7 +134,7 @@ hwloc__libxml_import_get_content(hwloc__xml_import_state_t state,
   if (!child || child->type != XML_TEXT_NODE) {
     if (expected_length)
       return -1;
-    *beginp = "";
+    *beginp = (char *) "";
     return 0;
   }
 
diff --git c/src/topology-xml-nolibxml.c w/src/topology-xml-nolibxml.c
index 147e703cc172..f11ae8cf5ff6 100644
--- c/src/topology-xml-nolibxml.c
+++ w/src/topology-xml-nolibxml.c
@@ -222,7 +222,7 @@ hwloc__nolibxml_import_get_content(hwloc__xml_import_state_t state,
   if (nstate->closed) {
     if (expected_length)
       return -1;
-    *beginp = "";
+    *beginp = (char *) "";
     return 0;
   }
 
diff --git c/src/topology-xml.c w/src/topology-xml.c
index 9eeb85b728a9..1f2a654a37a9 100644
--- c/src/topology-xml.c
+++ w/src/topology-xml.c
@@ -673,7 +673,7 @@ hwloc__xml_import_userdata(hwloc_topology_t topology __hwloc_attribute_unused, h
       }
 
   } else { /* always handle length==0 in the non-encoded case */
-      char *buffer = "";
+      char *buffer = (char *) "";
       if (length) {
 	ret = state->global->get_content(state, &buffer, length);
 	if (ret < 0)
